File: 2691-count-vowel-strings-in-ranges/2691-count-vowel-strings-in-ranges.cs
public class Solution {
    public int[] VowelStrings(string[] words, int[][] queries) {
        int n = words.Length, m = queries.Length;
        int[] pre = new int[n+1], ans = new int[m];
        int indx = 0;
        
        foreach(string u in words){
            pre[indx] = 0;
            int sz = u.Length;
            if((u[0] == 'a' || u[0] =='e' || u[0] =='i' || u[0] =='o' || u[0] =='u') && 
            (u[sz-1] == 'a' || u[sz-1] =='e' || u[sz-1] =='i' || u[sz-1] =='o' || u[sz-1] =='u') )
            {
                pre[indx] = 1;
             
            }
            if(indx > 0){
                pre[indx] += pre[indx-1];
               
            }
            indx++;
        }
        
        for(int i = 0 ; i < m ;i++){
            int x = queries[i][0];
            int y = queries[i][1];
            int res = pre[y];
            if(x > 0){
                res -= pre[x-1];
            }
            ans[i] = res;
        }
        return ans;
        
    }
}

File: 2691-count-vowel-strings-in-ranges/2691-count-vowel-strings-in-ranges.cpp
class Solution {
public:
    bool check(char ch){
        if(ch == 'a' || ch =='e' || ch =='i' || ch =='o' || ch == 'u'){
            return true;
        }
        return false;
    }
    vector<int> vowelStrings(vector<string>& words, vector<vector<int>>& queries) {
        int n = words.size(),indx = 0;
        vector<int> pre(n+1,0),ans;
        for(auto u:words){
            string s = u;
            if(check(s[0])){
                int sz = s.size();
                if(check(s[sz-1])){
                    pre[indx] = 1;
                }
            }
            if(indx > 0){
                pre[indx] += pre[indx-1]; 
            }
            indx++;
        }
        for(int i = 0 ; i<queries.size();i++){
            int x = queries[i][0];
            int y = queries[i][1];
            int res = 0;
            res = pre[y];
            if(x){
                res -= pre[x-1];
            }
            ans.push_back(res);
        }
        return ans;
    }
};